open-telemetry/src/main/java/com/datastax/oss/driver/internal/core/tracker/OtelRequestTracker.java

```diff
@@ -0,0 +1,466 @@
+   * db.collection.name 2. db.response.status_code
+   */
+  private static final AttributeKey<String> DB_SYSTEM_NAME =
+      AttributeKey.stringKey("db.system.name");
+
+  private static final AttributeKey<String> DB_NAMESPACE = AttributeKey.stringKey("db.namespace");
+  private static final AttributeKey<String> DB_OPERATION_NAME =
+      AttributeKey.stringKey("db.operation.name");
+  private static final AttributeKey<String> ERROR_TYPE = AttributeKey.stringKey("error.type");
+  private static final AttributeKey<Long> SERVER_PORT = AttributeKey.longKey("server.port");
+  private static final AttributeKey<String> CASSANDRA_CONSISTENCY_LEVEL =
+      AttributeKey.stringKey("cassandra.consistency.level");
+  private static final AttributeKey<String> CASSANDRA_COORDINATOR_DC =
+      AttributeKey.stringKey("cassandra.coordinator.dc");
+  private static final AttributeKey<String> CASSANDRA_COORDINATOR_ID =
+      AttributeKey.stringKey("cassandra.coordinator.id");
+  private static final AttributeKey<Long> CASSANDRA_PAGE_SIZE =
+      AttributeKey.longKey("cassandra.page.size");
+  private static final AttributeKey<Boolean> CASSANDRA_QUERY_IDEMPOTENT =
+      AttributeKey.booleanKey("cassandra.query.idempotent");
+  private static final AttributeKey<String> CASSANDRA_QUERY_ID =
+      AttributeKey.stringKey("cassandra.query.id");
+  private static final AttributeKey<Long> CASSANDRA_SPECULATIVE_EXECUTION_COUNT =
+      AttributeKey.longKey("cassandra.speculative_execution.count");
+  private static final AttributeKey<Long> DB_OPERATION_BATCH_SIZE =
+      AttributeKey.longKey("db.operation.batch.size");
+  private static final AttributeKey<String> DB_QUERY_TEXT = AttributeKey.stringKey("db.query.text");
+  private static final AttributeKey<String> SERVER_ADDRESS =
+      AttributeKey.stringKey("server.address");
+
+  public OtelRequestTracker(OpenTelemetry openTelemetry) {
+    this.tracer =
+        openTelemetry.getTracer("com.datastax.oss.driver.internal.core.tracker.OtelRequestTracker");
+    this.threadPool =
+        new LazyReference<>(
+            () ->
+                new ThreadPoolExecutor(
+                    1,
+                    Math.max(Runtime.getRuntime().availableProcessors(), 1),
+                    10,
+                    TimeUnit.SECONDS,
+                    new ArrayBlockingQueue<>(1000),
+                    new ThreadFactoryBuilder().setNameFormat("otel-thread-%d").build(),
+                    new ThreadPoolExecutor.AbortPolicy()));
+  }
+
+  @Override
+  public void close() throws Exception {
+    threadPool.get().shutdown();
+    threadPool.get().awaitTermination(10, TimeUnit.SECONDS);
+    logPrefixToTracingInfoMap.clear();
+  }
+
+  @Override
+  public void onRequestCreated(
+      @NonNull Request request,
+      @NonNull DriverExecutionProfile executionProfile,
+      @NonNull String requestLogPrefix) {
+    Span parentSpan = tracer.spanBuilder("Cassandra Java Driver Session Request").startSpan();
```

Comment:
Not sure if we should not add `.setParent(Context.current().with(Span.current()))` to link potential parent context started by the application.